packages/astro/src/core/sync/index.ts

```diff
@@ -1,62 +1,66 @@
 import { dim } from 'kleur/colors';
-import type fsMod from 'node:fs';
+import fsMod from 'node:fs';
 import { performance } from 'node:perf_hooks';
+import { fileURLToPath } from 'node:url';
 import { createServer, type HMRPayload } from 'vite';
-import type { Arguments } from 'yargs-parser';
-import type { AstroSettings } from '../../@types/astro';
+import type { AstroInlineConfig, AstroSettings } from '../../@types/astro';
 import { createContentTypesGenerator } from '../../content/index.js';
 import { globalContentConfigObserver } from '../../content/utils.js';
+import { telemetry } from '../../events/index.js';
+import { eventCliSession } from '../../events/session.js';
 import { runHookConfigSetup } from '../../integrations/index.js';
 import { setUpEnvTs } from '../../vite-plugin-inject-env-ts/index.js';
 import { getTimeStat } from '../build/util.js';
+import { resolveConfig } from '../config/config.js';
+import { createSettings } from '../config/settings.js';
 import { createVite } from '../create-vite.js';
 import { AstroError, AstroErrorData, createSafeError, isAstroError } from '../errors/index.js';
 import { info, type LogOptions } from '../logger/core.js';
-import { printHelp } from '../messages.js';
 
 export type ProcessExit = 0 | 1;
 
 export type SyncOptions = {
 	logging: LogOptions;
-	fs: typeof fsMod;
+	/**
+	 * Only used for testing
+	 * @internal
+	 */
+	fs?: typeof fsMod;
 };
 
-export async function syncCli(
-	settings: AstroSettings,
-	{ logging, fs, flags }: { logging: LogOptions; fs: typeof fsMod; flags?: Arguments }
+export async function sync(
```

Comment:
Will this be the official API? Just trying to understand where we are going

Reply:
Yes I think it would be nice to make this public. I also noticed the PR description had a typo. It's besides `check` and `add` that I had not refactored, not `sync` and `add` 😅 Will update that.

Reply:
Perfect, thank you. I am excited to see the final result!